diff -r 0c4b60ac68b4 Makefile
--- a/Makefile	Tue Jun 15 08:36:55 2010 -0400
+++ b/Makefile	Thu Sep 09 18:33:44 2010 +0100
@@ -1,23 +1,33 @@
 ERL          ?= erl
 APP          := webmachine
 
+EBIN_DIR=ebin
+SOURCE_DIR=src
+ERLC=erlc
+INCLUDE_DIR=include
+MOCHIWEB=../../../../rabbitmq-mochiweb/deps/mochiweb/mochiweb/
+ERLC_OPTS=-I include -pa $(MOCHIWEB)/ebin/ -o $(EBIN_DIR)
+INCLUDES=$(wildcard $(INCLUDE_DIR)/*.hrl)
+SOURCES=$(wildcard $(SOURCE_DIR)/*.erl)
+
+TARGETS=$(INCLUDE_DIR)/mochiweb/include/internal.hrl \
+	$(patsubst $(SOURCE_DIR)/%.erl, $(EBIN_DIR)/%.beam, $(SOURCES)) \
+	$(EBIN_DIR)/webmachine.app
+
 .PHONY: deps
 
-all: deps
-	@(./rebar compile)
+all: $(TARGETS)
 
-deps:
-	@(./rebar get-deps)
+$(EBIN_DIR)/%.beam: $(SOURCE_DIR)/%.erl
+	@mkdir -p $(EBIN_DIR)
+	$(ERLC) $(ERLC_OPTS) -pa $(EBIN_DIR) $<
+
+$(INCLUDE_DIR)/mochiweb/include/internal.hrl:
+	mkdir -p $(INCLUDE_DIR)/mochiweb/include
+	cp $(MOCHIWEB)/src/internal.hrl $(INCLUDE_DIR)/mochiweb/include
 
 clean:
-	@(./rebar clean)
+	rm -f $(EBIN_DIR)/*.beam
+	rm -rf $(INCLUDE_DIR)/mochiweb
 
 distclean: clean
-	@(./rebar delete-deps)
-
-edoc:
-	@$(ERL) -noshell -run edoc_run application '$(APP)' '"."' '[{preprocess, true},{includes, ["."]}]'
-
-test: all
-	scripts/run_tests.escript ebin | tee test.log
-
