diff -rNup scripts/rabbitmq-multi scripts-new/rabbitmq-multi
--- scripts/rabbitmq-multi	2008-12-17 18:38:14.000000000 +0000
+++ scripts-new/rabbitmq-multi	2008-12-24 18:13:15.000000000 +0000
@@ -30,7 +30,7 @@
 ##   Contributor(s): ______________________________________.
 ##
 
-[ -f /etc/default/rabbitmq ] && . /etc/default/rabbitmq
+[ -f /etc/conf.d/rabbitmq ] && . /etc/conf.d/rabbitmq
 
 [ "x" = "x$RABBITMQ_NODENAME" ] && RABBITMQ_NODENAME=${NODENAME}
 [ "x" = "x$RABBITMQ_NODENAME" ] && RABBITMQ_NODENAME=rabbit
diff -rNup scripts/rabbitmq-server scripts-new/rabbitmq-server
--- scripts/rabbitmq-server	2008-12-17 18:38:14.000000000 +0000
+++ scripts-new/rabbitmq-server	2008-12-24 18:13:15.000000000 +0000
@@ -30,7 +30,7 @@
 ##   Contributor(s): ______________________________________.
 ##
 
-[ -f /etc/default/rabbitmq ] && . /etc/default/rabbitmq
+[ -f /etc/conf.d/rabbitmq ] && . /etc/conf.d/rabbitmq
 
 [ "x" = "x$RABBITMQ_NODENAME" ] && RABBITMQ_NODENAME=${NODENAME}
 [ "x" = "x$RABBITMQ_NODENAME" ] && RABBITMQ_NODENAME=rabbit
