diff --git a/Makefile b/Makefile
index ae79ff2..ba0f710 100644
--- a/Makefile
+++ b/Makefile
@@ -1,29 +1,31 @@
+EBIN_DIR=ebin
+SOURCE_DIR=src
+ERLC=erlc
+INCLUDE_DIR=include
+ERLC_OPTS=-I include -o $(EBIN_DIR)
 
-PREFIX:=../
-DEST:=$(PREFIX)$(PROJECT)
+INCLUDES=$(wildcard $(INCLUDE_DIR)/*.hrl)
+SOURCES=$(wildcard $(SOURCE_DIR)/*.erl)
 
-REBAR=./rebar
+TARGETS=$(INCLUDE_DIR)/mochiweb/include/internal.hrl \
+	$(patsubst $(SOURCE_DIR)/%.erl, $(EBIN_DIR)/%.beam, $(SOURCES)) \
+	$(EBIN_DIR)/mochiweb.app
 
-all:
-	@$(REBAR) get-deps compile
-
-edoc:
-	@$(REBAR) doc
-
-test:
-	@rm -rf .eunit
-	@mkdir -p .eunit
-	@$(REBAR) skip_deps=true eunit
+all: $(TARGETS)
 
 clean:
-	@$(REBAR) clean
+	rm -f $(EBIN_DIR)/*.beam
+	rm -rf $(INCLUDE_DIR)/mochiweb
+
 
-build_plt:
-	@$(REBAR) build_plt
+$(EBIN_DIR)/%.beam: $(SOURCE_DIR)/%.erl
+	@mkdir -p $(EBIN_DIR)
+	$(ERLC) $(ERLC_OPTS) -pa $(EBIN_DIR) $<
 
-dialyzer:
-	@$(REBAR) analyze
 
-app:
-	@$(REBAR) create template=mochiwebapp dest=$(DEST) appid=$(PROJECT)
+$(INCLUDE_DIR)/mochiweb/include/internal.hrl:
+	mkdir -p $(INCLUDE_DIR)/mochiweb/include
+	cp $(SOURCE_DIR)/internal.hrl $(INCLUDE_DIR)/mochiweb/include
 
+$(EBIN_DIR)/mochiweb.app:
+	cp $(SOURCE_DIR)/mochiweb.app.src $(EBIN_DIR)/mochiweb.app
