diff -r 16a4000624a7 stomp/connect.py
--- a/stomp/connect.py	Sun May 02 18:15:34 2010 +0100
+++ b/stomp/connect.py	Thu Oct 07 11:04:59 2010 +0100
@@ -449,6 +449,9 @@
                 raise KeyError("Command %s requires header %r" % (command, required_header_key))
         self.__send_frame(command, headers, payload)
 
+    def send_frame(self, command, headers={}, payload=''):
+        self.__send_frame(command, headers, payload)
+        
     def __send_frame(self, command, headers={}, payload=''):
         """
         Send a STOMP frame.
